app/src/main/java/com/m/push/PushApplication.java

```diff
@@ -0,0 +1,32 @@
+package com.m.push;
+
+import android.app.Application;
+
+import org.jboss.aerogear.android.unifiedpush.PushRegistrar;
+import org.jboss.aerogear.android.unifiedpush.RegistrarManager;
+import org.jboss.aerogear.android.unifiedpush.fcm.AeroGearFCMPushJsonConfiguration;
+
+public class PushApplication extends Application {
+
+    private static final String PUSH_REGISTRAR_NAME = "myPushRegistrar";
+
+    @Override
+    public void onCreate() {
+        super.onCreate();
+
+        RegistrarManager.config(PUSH_REGISTRAR_NAME, AeroGearFCMPushJsonConfiguration.class)
+                .loadConfigJson(getApplicationContext())
+                .asRegistrar();
+
+    }
+
+    // Accessor method for Activities to access the 'PushRegistrar' object
+    public PushRegistrar getPushRegistrar() {
+        return RegistrarManager.getRegistrar(PUSH_REGISTRAR_NAME);
+    }
+
+    public void addMessage(String message){
```

Comment:
what's this function for? Is it just print out the message?

Reply:
Yes it only prints out the message. It is been called when a notification is received by the device no matter if the app is in the foreground or background. We can remove it.